Initial commit

giovannic opened the pull request:
This is a very minimal R wrapper for the P. Vivax model. It allows you to parameterise and run the default model and mosquito parameters through `run_simulation`.

Parameterising interventions and analysing the output is complicated and unspecified at the moment, so I've saved it for later. Michael's original code for parameterising interventions and analysing the output is in the R directory, but it's as yet untested.

Feel free to ignore files from the "man" and "docs" directories. They are autogenerated documentation by roxygen and pkgdown.

Commit messages:

* Model compiling
* Split run_simulation into args and running
* Tested model and parameterisation
* Remove compiled code
* readme documentation
* Make tests pass on check

richfitz commented on DESCRIPTION:
```diff
@@ -0,0 +1,31 @@
+  person(
+    given = "Michael",
+    family = "White",
+    role = c('aut', 'cre')
```
Somewhat counter intuitively, `cre` needs to go with the *maintainer* not really the creator. And because Michael does not have an email address you're getting the build error (though by the time this review is submitted you will likely have worked that out :) )

richfitz commented on NAMESPACE:
```diff
@@ -0,0 +1,3 @@
+useDynLib(vivax, .registration=TRUE)
+exportPattern("^[[:alpha:]]+")
```
Looks like this is not yet generated by roxygen (though there is at least the `@export` roxygen bits in the RcppAttributes generated code), and it would be better to move to explicit imports sooner rather than later.

giovannic commented on DESCRIPTION:
```diff
@@ -0,0 +1,26 @@
+Title: An R wrapper for a P. Vivax malaria transmission model
+Version: 0.0.1
+Date: 2020-01-31
+Authors@R: c(
```
That's a good spot. I'm going to try and add him as "author" and "creator". I've just emailed him to make sure it's ok to use his email.

richfitz commented on R/Intervention_file_generator.R:
```diff
@@ -0,0 +1,147 @@
+
+generate_intervention_file <- function() {
```
This function is going to write out the file `intervention_coverage.txt` into the current working directory: I think it should at least get a `path` argument to indicate the directory into which the file will be written

richfitz replied:
Actually, this does not appear to be (yet) used

giovannic replied:
I've made a note of this [here](https://github.com/mrc-ide/vivax/issues/3)

richfitz commented on R/Output_overview.R:
```diff
@@ -0,0 +1,268 @@
+                        "PvHR_batch_U10", "new_PCR_U10", "new_LM_U10", "new_D_U10", "new_T_U10",
+            "EIR", "LLIN_cov", "IRS_cov", "ACT_treat", "PQ_treat", "pregnant", "A_par", "A_clin")
+
+  par(ask=TRUE)
```
This plotting I think should be separated from the processing, and an interactively-stepped through series of plots seems like a suboptimal way of having to review things (the effect of `par(ask=TRUE)` is to request the user to step through each time a new plot is generated, in an interactive session).

Perhaps a simpler way to do this would be to translate this set of diagnostics into a small knitr report, store the Rmd for that into `inst/diagostics.Rmd` or similar, then compile a small report so that the effect of this function is to write out a html file - this could optionally displayed to the user in an interactive session too to capture more of the original intent.

In general though, functions should try not to alter the global state of the session, and calls to `par` that are not reset will do this.

@pwinskill probably has other ideas how this would fit into the malaria group workflows

pwinskill replied:
For our GTS use case we will mostly be aggregating up many runs before we summarise and visualise so I would not envisage much use for these built in plotting functions. 

I like the idea of the option for a automated diagnostic report to be produced if requested. I think this could be fairly simple (and might piggyback on some of the plotting functionality here). For example something like _Plot1: incidence over time, Plot2: prevalence over time, Plot3: prevalence~incidence and Plot4: intervention coverage over time_ would probably suffice to help us to spot most issues in individual simulations

giovannic replied:
Ok, it seems like this diagnostic code is not immediately useful so I've added a [issue ](https://github.com/mrc-ide/vivax/issues/5) in this milestone to remove it.

But some different diagnostics could potentially be useful, so I've added a [separate issue](https://github.com/mrc-ide/vivax/issues/6) which we can prioritise if the model starts bugging out.

giovannic commented on R/Intervention_file_generator.R:
```diff
@@ -0,0 +1,147 @@
+  ## At each timepoint, x% of the population receive a new net.
+  ## If they have an existing net it is replaced.
+
+  LLIN_years <- c() # c(2002, 2005, 2008, 2011, 2015, 2018, 2021, 2024, 2027, 2030, 2033, 2036, 2039, 2042)
```
Perfect, I have made an [issue](https://github.com/mrc-ide/vivax/issues/3) for this so that we action it. I've added it to the [GTS milestone](https://github.com/mrc-ide/vivax/milestone/1)

I've also created an issue for [c++ RNGs](https://github.com/mrc-ide/vivax/issues/2). If that's needed for GTS, please put it on the GTS milestone too and I'll have a look at it sooner rather than later.

richfitz commented on R/run.R:
```diff
@@ -0,0 +1,69 @@
+  punctulatus = NULL,
+  koliensis = NULL
+  ) {
+  basedir <- system.file('defaults', package = 'vivax')
```
The option `mustWork = TRUE` is useful in all uses of `system.file`, unfortunately

giovannic commented on R/Output_overview.R:
```diff
@@ -0,0 +1,268 @@
+
+present_output <- function(OUTPUT) {
```
Ah, if you could pop some examples in [this issue](https://github.com/mrc-ide/vivax/issues/4) that would be really helpful!

giovannic commented on R/run.R:
```diff
@@ -0,0 +1,69 @@
+
+#'@title Run the simulation
+#'@description parameterises and runs the model. The output will be returned
+#'@param model parameters to override
```
I've added some documentation and an example :)

giovannic commented:
@richfitz I've addressed the comments and made issues for the rest.

Travis is failing because of those cpp issues you mentioned, but I would really prefer to look at them in a different PR as you mentioned. I've made an [issue ](https://github.com/mrc-ide/vivax/issues/7) for that too.
